test/core/security/credentials_test.cc

```diff
@@ -2519,13 +2522,15 @@ TEST_F(TokenFetcherCredentialsTest, Basic) {
   // Next request will trigger a new fetch but will still use the
   // cached token.
   creds_->AddResult(MakeToken("bar"));
-  state = RequestMetadataState::NewInstance(
-      absl::OkStatus(), "authorization: foo", /*expect_delay=*/false);
+  LOG(INFO) << "Fourth request";
+  state =
+      RequestMetadataState::NewInstance(absl::OkStatus(), "authorization: foo");
```

Comment:
This change definitely looks wrong to me.  There should never be a delay here, because we should be using the cached token.  If we're not doing that, then there's a bug somewhere.

See #37531 for the code where I added this test, just recently.